mne_sandbox/preprocessing/_dss.py

```diff
@@ -53,27 +59,30 @@ def dss(data, data_max_components=None, data_thresh=0,
     References
     ----------
     .. [1] Särelä, Jaakko, and Valpola, Harri (2005). Denoising source
-    separation. Journal of Machine Learning Research 6: 233–72.
-
+       separation. Journal of Machine Learning Research 6: 233–72.
     .. [2] de Cheveigné, Alain, and Simon, Jonathan Z. (2008). Denoising based
-    on spatial filtering. Journal of Neuroscience Methods, 171(2): 331-339.
+       on spatial filtering. Journal of Neuroscience Methods, 171(2): 331-339.
     """
     if isinstance(data, (Epochs, EpochsArray)):
-        data_cov = compute_covariance(data).data
-        bias_cov = np.cov(data.average().pick_types(eeg=True, ref_meg=False).
-                          data)
+        epochs = data
         if return_data:
-            data = data.get_data()
+            data = epochs.get_data(picks='all')
     elif isinstance(data, np.ndarray):
         if data.ndim != 3:
             raise ValueError('Data to denoise must have shape '
                              '(n_trials, n_channels, n_times).')
-        data_cov = np.sum([np.dot(trial, trial.T) for trial in data], axis=0)
-        bias_cov = np.cov(data.mean(axis=0))
+        info = create_info(data.shape[1], 1000., 'eeg')
```

Comment:
As far as I can tell, `sfreq=1000.` and `ch_types='eeg'` are arbitrary choices here and don't affect the computation. Can you confirm?

Reply:
indeed